server/src/main/java/org/apache/uniffle/server/ShuffleTaskManager.java

```diff
@@ -411,18 +411,25 @@ public long requireBuffer(
       String appId, int shuffleId, List<Integer> partitionIds, int requireSize) {
     ShuffleTaskInfo shuffleTaskInfo = shuffleTaskInfos.get(appId);
     if (null == shuffleTaskInfo) {
-      return RequireBufferStatusCode.NO_REGISTER.statusCode();
+      throw new BufferLimitOfHugePartitionException("Buffer limit reached for huge partition");
```

Comment:
This is `NO_REGISTER`. So we shouldn't throw ` BufferLimitOfHugePartitionException` here.